cli/azd/internal/scaffold/funcs.go

```diff
@@ -14,6 +14,9 @@ import (
 func BicepName(name string) string {
 	sb := strings.Builder{}
 	separatorStart := -1
+
+	allUpper := isAllUpperCase(name)
```

Comment:
It's only 3 times slower (mem-alloc being part of it)

```
func isAllUpperCase(c string) bool {
	return c == strings.ToUpper(c)
}

func isAllUpperCaseNoAlloc(c string) bool {
	for i := range c {
		if 'a' <= c[i] && c[i] <= 'z' {
			return false
		}
	}

	return true
}

goos: darwin
goarch: arm64
pkg: github.com/azure/azure-dev/cli/azd/internal/scaffold
cpu: Apple M1 Max
Benchmark_isAllUpperCase-10    	 1284579	       979.6 ns/op	     141 B/op	       1 allocs/op
PASS
ok  	github.com/azure/azure-dev/cli/azd/internal/scaffold	2.471s


goos: darwin
goarch: arm64
pkg: github.com/azure/azure-dev/cli/azd/internal/scaffold
cpu: Apple M1 Max
Benchmark_isAllUpperCaseNoAlloc-10    	 3358752	       361.2 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/azure/azure-dev/cli/azd/internal/scaffold	1.839s
```

Reply:
> It's only 3 times slower

What input did you use to test it?  I would expect it to be about the same for the worst case in both (a word with all upper case), but faster when in char-by-char when there is at least one lower case. The sooner the lower case is found, the faster compared to calling `ToUpper()` 

This is cool. Thanks for looking into it.
My preference, as I mentioned before, is less code. But I'm happy eitherway

Reply:
I tested with random 32 character string.

Memory allocation is probably the bottleneck. I don't think I'd expect `ToUpper()` to ever be faster. It would require first allocation -- I think it at least happens twice as part of `string.Builder` as the buffer grows.

Then we'd walk the string once, converting to upper case.

Then the second walk to compare char by char. 

Unfortunately, when I see the algorithmic complexity behind the implementation, I end up thinking we should just make it "simple" -- and looping through char arrays is simpler for me.

Reply:
We can end up moving of these details into a `stringutil` package eventually. Some of these are just useful for general string processing, and perhaps if we see it from that perspective, we'd be more okay with using it.